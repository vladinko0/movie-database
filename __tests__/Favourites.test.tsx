import 'react-native';
import React from 'react';
import Favourites from '../src/screens/favourites';
import renderer from 'react-test-renderer';

jest.mock('@react-native-async-storage/async-storage', () => ({
  AsyncStorage: () => {},
}));
jest.mock('redux-persist', () => ({
  persistStore: () => {},
  persistReducer: () => {},
}));
jest.mock('@reduxjs/toolkit', () => ({
  configureStore: () => {},
  createSlice: () => {},
}));
jest.mock('../src/reduxStore', () => ({
  movieSlice: () => {},
}));
jest.mock('@react-navigation/native', () => ({
  useIsFocused: () => {},
}));

it('renders correctly Favourites screen', () => {
  const favourites = renderer.create(<Favourites />);
  expect(favourites).toMatchSnapshot();
});
