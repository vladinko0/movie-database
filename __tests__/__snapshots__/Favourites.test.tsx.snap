// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Favourites screen 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <BVLinearGradient
    colors={
      [
        4283866105,
        4289848060,
      ]
    }
    endPoint={
      {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      {
        "flex": 1,
        "paddingHorizontal": 20,
        "paddingTop": 20,
      }
    }
  >
    <RCTScrollView
      ListFooterComponent={[Function]}
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.7}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onLayout={[Function]}
        />
      </View>
    </RCTScrollView>
  </BVLinearGradient>
</View>
`;
